### ISSUE TITLE
make better decisions for minpoly based on domain

### ISSUE THREAD

**Issue by smichr** [sympy/sympy#14494]
#14382 gives
```
>>> minimal_polynomial(I, x, domain=QQ.algebraic_field(I))
x - I
```
instead of
```
x**2 + 1
```
but @jksuom noted: 
private methods like _minpoly_exp(ex, x) ... ignore the dom argument.

 @skirpichev suggested:
"...you could "sanitize" output of minpoly helpers: factorize one over specified dom (if it's badly supported) and select appropriate factor with _choose_factor()." Also, "[A more] systematic workaround may be using compose=False algorithm (currently, it doesn't support anything except Q) for algebraic extensions."

## example 
```

In [11]: minimal_polynomial(exp(I*pi/3), x, domain=QQ.algebraic_field(sqrt(3)*I))
Out[11]: 
 2        
x  - x + 1

In [12]: factor(_, extension=sqrt(3)*I)
Out[12]: 
⎛          ___  ⎞ ⎛          ___  ⎞
⎜    1   ╲╱ 3 ⋅ⅈ⎟ ⎜    1   ╲╱ 3 ⋅ⅈ⎟
⎜x - ─ - ───────⎟⋅⎜x - ─ + ───────⎟
⎝    2      2   ⎠ ⎝    2      2   ⎠
```

**Comment by jksuom** [comment]
I would try adding

    _, factors = factor_list(res, x, domain=dom)
    res = _choose_factor(factors, x, ex)

before the final `return res` in `_minpoly_compose`. Unit tests should then be added.

**Comment by abaokar-07** [comment]
> I would try adding
Still the output is `x - I` after the following change:
```
_, factors = factor_list(res, x, domain=dom)
res = _choose_factor(factors, x, ex)
return res
```

**Comment by skirpichev** [comment]
@abaokar-07, I hope someone here will "invent" the change, that "fix" this output ;)

**Comment by abaokar-07** [comment]
@skirpichev yeah sure, I just tried what @jksuom suggested and reported here

**Comment by skirpichev** [comment]
> yeah sure

Oh, I hope - not.

That was a joke, actually.  Please try to understand issue, before "fixing" it.  I.e. what is a correct output for the problem, what was incorrect and so on.

**Comment by abaokar-07** [comment]
> That was a joke, actually.

Yeah, I got that :D . And I'll keep in mind your suggestion, however, I tried to understand the issue, maybe I have misunderstood it
